updating to dynatrace-operator 1.3.0 and adding test job

akirasoft1 opened the pull request:
*Issue #, if available:*
227
*Description of changes:*
Updated to release 1.3.0 of our operator. cleaned up some api versions and added a testjob. The 10 min intervals might be a bit aggressive for running the job, most examples were only running once a day or so.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

elamaran11 commented:
Looks like your deployment `dynatrace-operator` is failing.
Specifically, it looks like these pods are failing: 
* Pod: `dynatrace-operator-6bf8447d8b-cwh68`.
---- 
Looks like your deployment `dynatrace-webhook` is failing.
Specifically, it looks like these pods are failing: 
* Pod: `dynatrace-webhook-6b784dd8b9-5l94d`.
* Pod: `dynatrace-webhook-6b784dd8b9-nfj8w`.
---- 
Looks like your deployment `dynatrace-webhook` is failing.
Specifically, it looks like these pods are failing: 
* Pod: `dynatrace-webhook-d64f58c9c-b2nzq`.
* Pod: `dynatrace-webhook-d64f58c9c-hrnk2`.
* Pod: `dynatrace-webhook-d64f58c9c-htkvj`.
----

akirasoft1 commented:
> Looks like your deployment `dynatrace-operator` is failing. Specifically, it looks like these pods are failing:
> 
> * Pod: `dynatrace-operator-6bf8447d8b-cwh68`.
> 
> Looks like your deployment `dynatrace-webhook` is failing. Specifically, it looks like these pods are failing:
> 
> * Pod: `dynatrace-webhook-6b784dd8b9-5l94d`.
> * Pod: `dynatrace-webhook-6b784dd8b9-nfj8w`.
> 
> Looks like your deployment `dynatrace-webhook` is failing. Specifically, it looks like these pods are failing:
> 
> * Pod: `dynatrace-webhook-d64f58c9c-b2nzq`.
> * Pod: `dynatrace-webhook-d64f58c9c-hrnk2`.
> * Pod: `dynatrace-webhook-d64f58c9c-htkvj`.

this is expected unless placeholder values in the yaml are replaced.

mikemcd3912 commented:
Hi @akirasoft1 , I'm running into a slight issue with the success of the Dynakube flux deployment not working based on the files in the addons repo I could use some help with:

![image](https://github.com/user-attachments/assets/3afdf220-95cd-43a3-8918-51f3489d5a14)

elamaran11 commented:
Looks like your deployment `dynatrace-operator` is failing.
Specifically, it looks like these pods are failing: 
* Pod: `dynatrace-operator-6bf8447d8b-jjqbl`.
---- 
Looks like your deployment `dynatrace-webhook` is failing.
Specifically, it looks like these pods are failing: 
* Pod: `dynatrace-webhook-d64f58c9c-ljjhg`.
* Pod: `dynatrace-webhook-d64f58c9c-sf67t`.
----

akirasoft1 commented:
> Hi @akirasoft1 , I'm running into a slight issue with the success of the Dynakube flux deployment not working based on the files in the addons repo I could use some help with:
> 
> ![image](https://private-user-images.githubusercontent.com/61101829/379428733-3afdf220-95cd-43a3-8918-51f3489d5a14.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk3MTcyMzcsIm5iZiI6MTcyOTcxNjkzNywicGF0aCI6Ii82MTEwMTgyOS8zNzk0Mjg3MzMtM2FmZGYyMjAtOTVjZC00M2EzLTg5MTgtNTFmMzQ4OWQ1YTE0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDIzVDIwNTUzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFlNjkyOTZhNjkyNGJjYmY5Y2Q0YmVkYTY4MjUyODk3OTZjMmQ1ZTJmN2I3YmE5NTgzNGNlYTM1ODhkZjIzMDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.AvN7kZwqf5H30af3KSeS42ELYslLnIhDRgL8uIxqgLk)

that is expected without replacing placeholder values in the variables.yaml with valid ones (in addition to the creds the operator itself is configured to fetch from secrets there is the apiurl and dynakubename that aren't secrets per se but should not be shared.) you can email me at michael.villiger @ dynatrace.com and I can exchange the

mikemcd3912 commented on eks-anywhere-common/Addons/Partner/dynatrace/dynatrace-variables.yaml:
There appears to be an issue with the apiurl replaceme since flux doesn't want to accept <>, please update the placeholder according to the acceptable values in the screenshot: 
![image](https://github.com/user-attachments/assets/f12f6299-dcf9-4c1e-8792-32bdad41a9f0)

akirasoft1 replied:
so replaceme will fail as well. It needs to be replaced with a _valid_ value.

mikemcd3912 replied:
Yep correct - We may want to replace this with a secret if the endpoints/creds needs to be shielded from the public repo

mikemcd3912 reviewed (commented):
@akirasoft1 Updates look good but there appears to be an issue with a placeholder not working for Flux and we'll need the value in there to work as-is for the GitOps to work in our ongoing testing. I think we may want to set that up on a secret if the value shouldn't be in the public facing repo

elamaran11 commented on eks-anywhere-common/Addons/Partner/dynatrace/dynatrace-variables.yaml:
```diff
@@ -6,8 +6,8 @@ metadata:
 data:
   # api url for dynatrace environment 
-  apiurl: "<replaceme>"
+  apiurl: "https://fjy55252.sprint.apps.dynatracelabs.com/api"
```
Is this a permanent URL? We cant have temporary URLs for this continuous testing.

elamaran11 reviewed (commented):
@akirasoft1 Question on your change

elamaran11 commented:
Looks like your deployment `dynatrace-operator` is failing.
Specifically, it looks like these pods are failing: 
* Pod: `dynatrace-operator-6bf8447d8b-jkwnc`.
---- 
Looks like your deployment `dynatrace-webhook` is failing.
Specifically, it looks like these pods are failing: 
* Pod: `dynatrace-webhook-d64f58c9c-gt4g5`.
----

akirasoft1 commented:
i added the values back into the values file for now. You will need to reach out to me for the secrets values that you will load on your secret store...

elamaran11 commented:
> i added the values back into the values file for now. You will need to reach out to me for the secrets values that you will load on your secret store...

@akirasoft1 We cannot have hardcoded values in the variables. You can share us the secret via email which can be permanent and create ExternalSecret resource which you should reference in the values. Once you share the secret, We can create Secrets in Secrets Manager which ExternalSecret can reference and render. You can check with our other partner submissions.

akirasoft1 commented:
> > i added the values back into the values file for now. You will need to reach out to me for the secrets values that you will load on your secret store...
> 
> @akirasoft1 We cannot have hardcoded values in the variables. You can share us the secret via email which can be permanent and create ExternalSecret resource which you should reference in the values. Once you share the secret, We can create Secrets in Secrets Manager which ExternalSecret can reference and render. You can check with our other partner submissions.

yes, all the above is how it was before, I basically reverted changes to the file and updated to a more permanent URL. if someone wants to tell me how to reference secrets inside these fluxcd things i can copy paste?

elamaran11 commented:
> > > i added the values back into the values file for now. You will need to reach out to me for the secrets values that you will load on your secret store...
> > 
> > 
> > @akirasoft1 We cannot have hardcoded values in the variables. You can share us the secret via email which can be permanent and create ExternalSecret resource which you should reference in the values. Once you share the secret, We can create Secrets in Secrets Manager which ExternalSecret can reference and render. You can check with our other partner submissions.
> 
> yes, all the above is how it was before, I basically reverted changes to the file and updated to a more permanent URL. if someone wants to tell me how to reference secrets inside these fluxcd things i can copy paste?

You can check [this](https://github.com/aws-samples/eks-anywhere-addons/tree/main/eks-anywhere-common/Addons/Partner/Perfectscale) example or any other example in partner to understand how to handle ExternalSecrets.

elamaran11 commented:
Looks like your deployment `dynatrace-operator` is failing.
Specifically, it looks like these pods are failing: 
* Pod: `dynatrace-operator-6bf8447d8b-jkwnc`.
---- 
Looks like your deployment `dynatrace-webhook` is failing.
Specifically, it looks like these pods are failing: 
* Pod: `dynatrace-webhook-d64f58c9c-gt4g5`.
----

akirasoft1 commented:
> > > > i added the values back into the values file for now. You will need to reach out to me for the secrets values that you will load on your secret store...
> > > 
> > > 
> > > @akirasoft1 We cannot have hardcoded values in the variables. You can share us the secret via email which can be permanent and create ExternalSecret resource which you should reference in the values. Once you share the secret, We can create Secrets in Secrets Manager which ExternalSecret can reference and render. You can check with our other partner submissions.
> > 
> > 
> > yes, all the above is how it was before, I basically reverted changes to the file and updated to a more permanent URL. if someone wants to tell me how to reference secrets inside these fluxcd things i can copy paste?
> 
> You can check [this](https://github.com/aws-samples/eks-anywhere-addons/tree/main/eks-anywhere-common/Addons/Partner/Perfectscale) example or any other example in partner to understand how to handle ExternalSecrets.

we were using external secrets already for populating a secret the operator needed. i've sent an email, maybe this will work?

mikemcd3912 commented on eks-anywhere-common/Addons/Partner/dynatrace/dynakube-operator.yaml:
It appears I'm still getting an error with the replacement of this secret value conflicting with Flux - I have it loaded and the creation was successful: 
![image](https://github.com/user-attachments/assets/105b34bc-f2b7-4fa0-937f-eb855493f7f1)

elamaran11 commented:
Looks like your deployment `dynatrace-dynatrace-operator` is failing.
Specifically, it looks like these pods are failing: 
* Pod: `dynatrace-dynatrace-operator-6d6f5855d5-sncs9`.
---- 
Looks like your deployment `dynatrace-operator` is failing.
Specifically, it looks like these pods are failing: 
* Pod: `dynatrace-operator-6bf8447d8b-qdblc`.
---- 
Looks like your deployment `dynatrace-webhook` is failing.
Specifically, it looks like these pods are failing: 
* Pod: `dynatrace-webhook-c989b778c-ms29k`.
----

elamaran11 commented:
Looks like your deployment `dynatrace-dynatrace-operator` is failing.
Specifically, it looks like these pods are failing: 
* Pod: `dynatrace-dynatrace-operator-6d6f5855d5-sncs9`.
---- 
Looks like your deployment `dynatrace-operator` is failing.
Specifically, it looks like these pods are failing: 
* Pod: `dynatrace-operator-6bf8447d8b-qdblc`.
---- 
Looks like your deployment `dynatrace-webhook` is failing.
Specifically, it looks like these pods are failing: 
* Pod: `dynatrace-webhook-c989b778c-ms29k`.
----

akirasoft1 commented:
I'm going to kill this PR and open a new one as I have moved to v1.3.2 of the dynatrace operator and I believe I have something that works now.

elamaran11 commented:
Looks like your deployment `dynatrace-operator` is failing.
Specifically, it looks like these pods are failing: 
* Pod: `dynatrace-operator-56b6c4c694-gnlck`.
---- 
Looks like your deployment `dynatrace-webhook` is failing.
Specifically, it looks like these pods are failing: 
* Pod: `dynatrace-webhook-795bcc9b47-6fdqn`.
* Pod: `dynatrace-webhook-795bcc9b47-792xf`.
----
